### DISCUSSION TITLE
Problems with starting Postgres after update

### DISCUSSION THREAD

**Discussion by mtaponen** [Nico640/docker-unms#99] [General]
Firstly - thanks for this repo and the work you people are putting in to this! It is awesome.
Then - to give something back to the project I will share a problem and a fix in order to help anyone else struggling with this. 

You can also cheat and hop to the end.

1) The setup
I have Raspberry Pi  running all my 24/7 docker stuff. I mostly use NAS as the persistent disk to avoid any issues with the SD card. My docker version is currently Docker version 20.10.22
2) The original problem
I have been running the UNMS/UISP for couple of years now and one day it had stopped working. 
container logs show:

```
2021/11/16 20:35:02 [error] 16009#16009: *643 connect() failed (111: Connection refused) while connecting to upstream, client: 192.168.11.5, server: , request: "GET /nms/api/v2.1/nms/heartbeat HTTP/1.1", upstream: "http://127.0.0.1:8081/nms/api/v2.1/nms/heartbeat", host: "192.168.11.80", referrer: "https://192.168.11.80/nms/settings/backup"
ERROR:  Wrong key or corrupt data
CONTEXT:  SQL function "unms_decrypt" statement 1
STATEMENT:  SELECT unms_decrypt(unms_encrypt('unms-rulez')::text, 'v>q+{JVmN$|7vFx[4fxM')
ERROR:  Wrong key or corrupt data
CONTEXT:  SQL function "unms_decrypt" statement 1
```
After some investigation and not finding any remedy I decided to update my ´docker-unms´ docker image to the latest version.
https://github.com/Nico640/docker-unms/releases/tag/2.3.57 was the latest when I pulled.

3) Clean install
Rather than updating the datafiles I decided to do a clean install. So I took the config-directory on the host -raspberry and renamed it by `mv config config-backup && mkdir config && chown 911:911 config`
Then I started the container by running `docker-compose up -d``

4) The actual issue
Then I noticed that UISP would not come up. It woudl ask me to wait and then after a while it reported a problem
From the Container Logs I see:

```

ESC[36muisp-unms-1 |ESC[0m Starting nginx...
ESC[36muisp-unms-1 |ESC[0m Running entrypoint.sh
ESC[36muisp-unms-1 |ESC[0m Creating nginx configuration
ESC[36muisp-unms-1 |ESC[0m /run/postgresql:5432 - no response
ESC[36muisp-unms-1 |ESC[0m Waiting for postgres to come up...
ESC[36muisp-unms-1 |ESC[0m Enabling UNMS https and wss connections on port 443
ESC[36muisp-unms-1 |ESC[0m Will use existing SSL certificate
ESC[36muisp-unms-1 |ESC[0m 2023-12-11 13:56:19.292 EET [2655] LOG:  could not translate host name "localhost", service "5432" to addr
ess: Try again
ESC[36muisp-unms-1 |ESC[0m 2023-12-11 13:56:19.292 EET [2655] WARNING:  could not create listen socket for "localhost"
ESC[36muisp-unms-1 |ESC[0m 2023-12-11 13:56:19.292 EET [2655] FATAL:  could not create any TCP/IP sockets
ESC[36muisp-unms-1 |ESC[0m 2023-12-11 13:56:19.301 EET [2655] LOG:  database system is shut down
ESC[36muisp-unms-1 |ESC[0m Entrypoint finished

```

5) the fix
After mucking about I tried to change the `postgresql.conf`
```
# listen_addresses = 'localhost'                # what IP address(es) to listen on;
listen_addresses = '127.0.0.1'
```

Now I got the UISP to start. Not sure if I get the recovery working but that's a different story...

**TL;DR: When running image tagged [2.3.57](https://github.com/Nico640/docker-unms/releases/tag/2.3.57)  on Docker version 20.10.22 on Raspberry Pi I could not make clean start after installation.
I edited `postgresql.conf` to have `127.0.0.1` as the hostname and that seeminly fixed the issue**

**Comment by mtaponen** [comment]
And to follow up - seems that the recovery was working when done from withtin working (blank) UISP. The first-greeting-login-recovery lead to unfunctional UISP (for some reason unknown).
